src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -231,6 +231,166 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::CreateNew
     return *newProfile;
 }
 
+// Method Description:
+// - Duplicate a new profile based off another profile's settings
+// Return Value:
+// - a reference to the new profile
+winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::DuplicateProfile(Model::Profile source)
+{
+    auto duplicated{ _userDefaultProfileSettings->CreateChild() };
+    _allProfiles.Append(*duplicated);
+
+    const auto newName{ fmt::format(L"{} ({})", source.Name(), RS_(L"CopySuffix")) };
+    duplicated->Name(winrt::hstring(newName));
+

```

Comment:
We do not manually assign any guid here, later on a guid is automatically generated for the duplicated profile from its name

Reply:
Maybe we should, to avoid the (incredibly rare but absolutely _possible_) case from above? I think it'd be totally unexpected if you duplicated "Foo" and ended up in the settings for "Bar"

Reply:
I'm not sure I agree that this is on us to solve. If we're going as far as to consider what if the user manually inputs a guid that will conflict with a guid we generate, then couldn't they achieve the same thing by simply manually assigning the same guid to two different profiles in their settings file? It feels like one of those 'do bad things get bad results' that I'm not sure we should try to work around

Reply:
Meh, I'm fine with that for now. Just so long as we're aware of it 😉